src/blockbandedcholesky.jl

```diff
@@ -0,0 +1,91 @@
+
+
+sparsecholesky(A::Symmetric{<:Real,<:BlockArray},
+::Val{false}=Val(false); check::Bool = true) = sparsecholesky!(cholcopy(A); check = check)
+
+
+
+function _blockbandedcholesky!(A::BlockArray{T}, ::Type{UpperTriangular}) where T<:Real
+    n = blocksize(A)[1]
+    k_end = 0
+
+    @inbounds begin
+        for i = 1:n
+            Pii = getblock(A,i,i)
+            for k = 1:i-1
```

Comment:
This should loop over `colsupport(A.blocks, i) ∩ 1:i-1` I think

Reply:
> colsupport(_, A, j) = axes(A,1)
> 
> 
> """"
>     colsupport(A, j)
> 
> gives an iterator containing the possible non-zero entries in the j-th column of A.
> """
> colsupport(A, j) = colsupport(MemoryLayout(typeof(A)), A, j)

The code of `colsupport(A, j)`  is just the `axes(A,1)`, so this may not be useful.

Reply:
It's overloaded for banded matrices: https://github.com/JuliaMatrices/BandedMatrices.jl/blob/3eb9c0d77f471db67b87b7dc5f9832c1d0beb0b2/src/generic/AbstractBandedMatrix.jl#L89

Reply:
Since we only care about the blockbanded cases, maybe we can move the file into the package BlockBandedMatrices?
Then it will be convenient where all the prerequisite functions  are included.

Reply:
No, it should be here. It will do the right thing for both dense and banded formats.

Reply:
But if we want to use `colsupport()`,  the matrix type should be either `BandedMatrix` or `BlockBandedMatrix`. 
Or we can overload `colsupport()` and assume without losing generality that `BlockArray` is the dense case and others are banded cases?

Reply:
Another thing is that why `LAPACK.potrf!()` do not work in-place on BlockBandedMatrices? 
It gives the right output but the block will not be replaced by the decomposed matrix.